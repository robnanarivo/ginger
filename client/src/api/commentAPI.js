import axios from 'axios';
import domain from '../constants/profile';

const createComment = async (comment) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.post(`${domain}/api/comments/`, comment, { headers: config });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const deleteComment = async (commentId) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.delete(`${domain}/api/comments/${commentId}`, { headers: config });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const updateComment = async (commentId, comment) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.put(`${domain}/api/comments/${commentId}`, comment, { headers: config });
    return res.data;
  } catch (err) {
    if (err.response) {
      return err.response.data;
    }
    return { error: 'Cannot connect to server' };
  }
};

const getPost = async (postId) => {
  try {
    const res = await axios.get(`${domain}/api/posts/${postId}`);
    const { data } = res;
    const { content } = data;
    if (content.audio) {
      Object.assign(content, { audio: `${domain}/api/multimedia/${content.audio}` });
    }
    if (content.video) {
      Object.assign(content, { video: `${domain}/api/multimedia/${content.video}` });
    }
    if (content.image) {
      Object.assign(content, { image: `${domain}/api/multimedia/${content.image}` });
    }
    return data;
  } catch (err) {
    return err.response;
  }
};

export {
  createComment,
  deleteComment,
  updateComment,
  getPost,
};
