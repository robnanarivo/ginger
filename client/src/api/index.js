import {
  getListOfGroups,
  getGroupData,
  updateGroup,
  removeAdmin,
  getGroupAdmins,
  createGroup,
} from './fetcher';
import login from './login';
import register from './register';
import { getNotifications, updateNotification } from './notificationAPI';
import getUsers from './user';
import { updateProfile, getProfile } from './profileAPI';
import {
  createComment, deleteComment, updateComment, getPost,
} from './commentAPI';

export {
  getGroupData,
  getListOfGroups,
  updateGroup,
  getGroupAdmins,
  removeAdmin,
  login,
  register,
  createGroup,
  getNotifications,
  updateNotification,
  getUsers,
  updateProfile,
  getProfile,
  createComment,
  deleteComment,
  updateComment,
  getPost,
};
