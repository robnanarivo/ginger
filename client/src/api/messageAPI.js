import axios from 'axios';
import domain from '../constants/profile';

const sendMessage = async (message) => {
  try {
    const res = await axios.post(`${domain}/api/messages`, message);
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const getMessageHistory = async (chatter1, chatter2, offset) => {
  const params = {
    chatter1,
    chatter2,
    offset,
    limit: 10,
  };
  try {
    const res = await axios.get(`${domain}/api/messages`, { params });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

const getNewMessage = async (chatter1, chatter2) => {
  const params = {
    chatter1,
    chatter2,
    offset: 0,
    limit: 1,
  };
  try {
    const res = await axios.get(`${domain}/api/messages`, { params });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

export {
  sendMessage,
  getMessageHistory,
  getNewMessage,
};
