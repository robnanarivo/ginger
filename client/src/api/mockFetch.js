export const groupData = {
  topics: ['Sports', 'Universe', 'Science', 'Astronomy'],
  groupInfo: { createOn: 'Feb 20 2008', member: '2.8k' },
  groupId: '11',
  admins: [],
  users: [],
  posts: [
    {
      postId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      groupId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      content: {
        title: 'Stars are so beautiful',
        text: 'star!',
        image: '/starPic.jpeg',
        audio: '',
        video: '',
      },
      date: 'September 16, 2021',
      commentCount: 30,
      likeCount: 8750,
      avatar: 'AB',
      timeStamp: {},
      comments: [
        {
          commentId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          content: 'somewhat random text',
          timeStamp: {},
          parentPostId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
        },
      ],
      flagedBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
      hiddenBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
    },
    {
      postId: 'd290f1ee-6c54-4b01-90e6-d701748f0852',
      groupId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      content: {
        title: 'Look at the Galaxy!',
        text: 'beautiful night',
        image: '/galaxyPic.jpeg',
        audio: '',
        video: '',
      },
      timeStamp: {},
      date: 'September 17, 2021',
      commentCount: 328,
      likeCount: 18650,
      avatar: 'NK',
      comments: [
        {
          commentId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          content: 'somewhat random text',
          timeStamp: {},
          parentPostId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
        },
      ],
      flagedBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
      hiddenBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
    },
  ],
  groupName: 'Galaxy',
  userStatus: 'joined',
};

export const listOfAdmins = [
  {
    userId: '11',
    userName: 'Alex',
    profilePicture: '',
  },
  {
    userId: '12',
    userName: 'Tom',
    profilePicture: '',
  },
];

export const listOfGroupAll = [
  {
    groupId: '11',
    groupName: 'Galaxy',
    groupPic: '',
    member: '40',
    userStatus: 'joined',
  },
  {
    groupId: '2',
    groupName: 'SEPTA',
    groupPic: '',
    member: '40',
    userStatus: 'not joined',
  },
  {
    groupId: '3',
    groupName: 'Study Group',
    groupPic: '',
    member: '48920',
    userStatus: 'joined',
  },
  {
    groupId: '4',
    groupName: 'Involution',
    groupPic: '',
    member: '345781',
    userStatus: 'pending',
  },
  {
    groupId: '5',
    groupName: 'Ubuntu',
    groupPic: '',
    member: '908907',
    userStatus: 'pending',
  },
  {
    groupId: '6',
    groupName: 'Debian',
    groupPic: '',
    member: '462',
    userStatus: 'not joined',
  },
  {
    groupId: '7',
    groupName: 'CentOS',
    groupPic: '',
    member: '471',
    userStatus: 'joined',
  },
  {
    groupId: '8',
    groupName: 'Assembly',
    groupPic: '',
    member: '589',
    userStatus: 'not joined',
  },
  {
    groupId: '9',
    groupName: 'JIT',
    groupPic: '',
    member: '25',
    userStatus: 'not joined',
  },
  {
    groupId: '10',
    groupName: 'NLP',
    groupPic: '',
    member: '5',
    userStatus: 'not joined',
  },
];

export const listOfGroupPub = [
  {
    groupId: '11',
    groupName: 'Galaxy',
    groupPic: '',
    groupInfo: { createOn: '', member: '40' },
    userStatus: 'joined',
  },
  {
    groupId: '2',
    groupName: 'SEPTA',
    groupPic: '',
    groupInfo: { createOn: '', member: '3640' },
    userStatus: 'not join',
  },
  {
    groupId: '3',
    groupName: 'Study Group',
    groupPic: '',
    groupInfo: { createOn: '', member: '48920' },
    userStatus: 'joined',
  },
  {
    groupId: '4',
    groupName: 'Involution',
    groupPic: '',
    groupInfo: { createOn: '', member: '345781' },
    userStatus: 'pending',
  },
  {
    groupId: '5',
    groupName: 'Ubuntu',
    groupPic: '',
    groupInfo: { createOn: '', member: '908907' },
    userStatus: 'pending',
  },
  {
    groupId: '6',
    groupName: 'Debian',
    groupPic: '',
    groupInfo: { createOn: '', member: '462' },
    userStatus: 'notJoined',
  },
];
