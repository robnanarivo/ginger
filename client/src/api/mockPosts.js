const postData = {
  topics: ['Sports', 'Universe', 'Science', 'Astronomy'],
  groupInfo: { createOn: 'Feb 20 2008', member: '2.8k' },
  groupId: '11',
  posts: [
    {
      postId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      groupId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
      creatorId: '61b26b150c01529bed74c6f0',
      creatorName: 'Wang',
      title: 'Stars are so beautiful',
      content: {
        text: 'star!',
        image: '/starPic.jpeg',
        audio: '',
        video: '',
      },
      profilePicture: '/starPic.jpeg',
      timestamp: 1639429592291,
      comments: [
        {
          commentId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          userId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
          content: 'somewhat random text',
          timeStamp: 1639429592456,
          parentPostId: 'd290f1ee-6c54-4b01-90e6-d701748f0851',
        },
      ],
      flagedBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
      hiddenBy: [
        'd290f1ee-6c54-4b01-90e6-d701748f0851',
      ],
    },
  ],
  groupName: 'Galaxy',
  userStatus: 'joined',
};

export default postData;
