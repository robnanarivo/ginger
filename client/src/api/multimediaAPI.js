import axios from 'axios';
import domain from '../constants/profile';

const upload = async (file) => {
  try {
    const formData = new FormData();
    formData.append('file', file);
    const res = await axios.post(`${domain}/api/multimedia`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return res.data;
  } catch (err) {
    return err.response;
  }
};

export default upload;
