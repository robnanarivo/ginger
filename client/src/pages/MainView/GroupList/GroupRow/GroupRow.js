import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import Box from '@mui/material/Box';
import Avatar from '@mui/material/Avatar';
import FolderIcon from '@mui/icons-material/Folder';
import Button from '@mui/material/Button';
import { Typography } from '@mui/material';
import { updateGroup } from '../../../../api';

function displayJoinButton(updateUS, userStatus, groupId) {
  if (userStatus === 'joined') {
    return (
      <Button
        variant="outlined"
        sx={{ marginLeft: 5, minWidth: '100px', marginRight: 5 }}
        onClick={() => {
          const res = updateGroup(groupId, 'leave', [sessionStorage.getItem('userId')], []);
          if (res) {
            updateUS('not joined');
          }
        }}
      >
        LEAVE
      </Button>
    );
  }
  if (userStatus === 'pending') {
    return (
      <Button
        variant="contained"
        sx={{
          marginLeft: 5,
          minWidth: '100px',
          marginRight: 5,
          color: '#AFAFAF',
          backgroundColor: '#E0E0E0',
          '&:hover': {
            cursor: 'pointer',
            backgroundColor: '#C0C0C0',
          },
        }}
        onClick={() => {
          const res = updateGroup(groupId, 'leave', [sessionStorage.getItem('userId')], []);
          if (res) {
            updateUS('not joined');
          }
        }}
      >
        PENDING
      </Button>
    );
  }
  if (userStatus === 'not joined') {
    return (
      <Button
        variant="contained"
        sx={{ marginLeft: 5, minWidth: '100px', marginRight: 5 }}
        onClick={() => {
          const res = updateGroup(groupId, 'apply', [sessionStorage.getItem('userId')], []);
          if (res) {
            updateUS('pending');
          }
        }}
      >
        JOIN
      </Button>
    );
  }
  return (<div />);
}

export default function GroupRow(props) {
  const history = useHistory();
  const {
    groupId,
    number,
    groupName,
    groupPic,
    member,
    userStatus,
  } = props;
  const [uS, setUS] = useState(userStatus);
  const updateUS = (us) => {
    setUS(us);
  };
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        p: 1,
        height: '60px',
      }}
    >
      <Box
        onClick={() => { history.push(`/groupboard/${groupId}`); }}
        sx={{
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          p: 1,
          width: '70%',
          '&:hover': {
            cursor: 'pointer',
          },
        }}
        data-testid="b1"
      >
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
            p: 1,
            minWidth: '150px',
          }}
        >
          <Typography sx={{ marginLeft: 5 }}>
            {number}
          </Typography>
          <Box sx={{ flexGrow: 1 }} />
          <Avatar>
            {groupPic === ''
              ? (<FolderIcon />)
              : (<Avatar alt="Group Logo" src={groupPic} sx={{ backgroundColor: '#FFFFFF' }} />)}
          </Avatar>
        </Box>
        <Typography sx={{ marginLeft: 5 }}>
          {groupName}
        </Typography>
        <Box sx={{ flexGrow: 1 }} />
        {
          member > 1
            ? (
              <Typography>
                {`${member} Members`}
              </Typography>
            ) : (
              <Typography>
                {`${member} Member`}
              </Typography>
            )
        }

      </Box>
      <Box sx={{ flexGrow: 1 }} />
      {displayJoinButton(updateUS, uS, groupId)}
    </Box>
  );
}
