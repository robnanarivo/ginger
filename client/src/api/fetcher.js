// import { groupData, listOfGroupAll, listOfGroupPub } from './mockFetch';
import axios from 'axios';
import { listOfAdmins } from './mockFetch';

const domain = !process.env.NODE_ENV || process.env.NODE_ENV === 'development'
  ? 'http://localhost:5000'
  : '';

const getGroupData = async (groupId) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const ret = await axios.get(`${domain}/api/group/${groupId}`, { headers: config });
    const { data } = ret;
    if (data.groupIcon !== '') {
      Object.assign(data, { groupIcon: `${domain}/api/multimedia/${data.groupIcon}` });
    } else {
      Object.assign(data, { groupIcon: '' });
    }
    // data.posts.forEach((post) => {
    //   const { content } = post;
    //   if (content.audio) {
    //     Object.assign(content, { audio: `${domain}/api/multimedia/${content.audio}` });
    //   }
    //   if (content.video) {
    //     Object.assign(content, { video: `${domain}/api/multimedia/${content.video}` });
    //   }
    //   if (content.image) {
    //     Object.assign(content, { image: `${domain}/api/multimedia/${content.image}` });
    //   }
    // });
    return data;
  } catch (err) {
    return err.response;
  }
};

const createGroup = async (groupInfo) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.post(`${domain}/api/group/`, groupInfo, { headers: config });
    return res.data;
  } catch (err) {
    // console.log(err);
    return err.response.data;
  }
};

const updateGroup = async (groupId, action, userIds, adminIds) => {
  const token = sessionStorage.getItem('userToken');
  const updateInfo = { action, userIds, adminIds };
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const res = await axios.put(`${domain}/api/group/${groupId}`, updateInfo, { headers: config });
    // console.log(res);
    if (res.data.results === 'success') {
      return true;
    }
    return false;
  } catch (err) {
    // console.log(err);
    return false;
  }
};

const getListOfGroups = async (topic) => {
  const token = sessionStorage.getItem('userToken');
  try {
    const config = {
      authorization: `Bearer ${token}`,
    };
    const ret = await axios.get(`${domain}/api/groups/${topic}`, { headers: config });
    const { data } = ret;
    data.forEach((d) => {
      if (d.groupIcon !== '') {
        Object.assign(d, { groupIcon: `${domain}/api/multimedia/${d.groupIcon}` });
      } else {
        Object.assign(d, { groupIcon: '' });
      }
    });
    return ret.data;
  } catch (err) {
    // console.log(err.response);
    return err.response;
  }
};

// const getGroupAdmins = async (groupId) => {
//   const groupInfo = ['Alex', 'Tom'];
//   if (groupId) {
//     return listOfAdmins;
//   }
//   // const token = sessionStorage.getItem('userToken');
//   // try {
//   //   const config = {
//   //     authorization: `Bearer ${token}`,
//   //   };
//   //   const ret = await axios.get(`${domain}/api/group/${groupId}`, { headers: config });
//   //   const { data } = ret;
//   //   console.log(data);
//   // } catch (err) {
//   //   return err.response;
//   // }
//   return groupInfo;
// };

const getGroupAdmins = async (groupId) => {
  const groupInfo = ['Alex', 'Tom'];
  if (groupId) {
    return listOfAdmins;
  }
  // const token = sessionStorage.getItem('userToken');
  // try {
  //   const config = {
  //     authorization: `Bearer ${token}`,
  //   };
  //   const ret = await axios.get(`${domain}/api/group/${groupId}`, { headers: config });
  //   const { data } = ret;
  //   console.log(data);
  // } catch (err) {
  //   return err.response;
  // }
  return groupInfo;
};

const removeAdmin = async (adminId) => {
  if (!adminId) {
    return true;
  }
  return true;
};

export {
  getGroupData,
  getListOfGroups,
  createGroup,
  updateGroup,
  getGroupAdmins,
  removeAdmin,
};
