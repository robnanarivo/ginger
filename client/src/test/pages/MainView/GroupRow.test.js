import React from 'react';
import renderer from 'react-test-renderer';
import {
  screen,
  render,
  act,
} from '@testing-library/react';
import { Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import GroupRow from '../../../pages/MainView/GroupList/GroupRow';
import { listOfGroupAll } from '../../../api/mockFetch';
import { updateGroup } from '../../../api';
import 'jest-styled-components';

jest.mock('../../../api');

describe('GroupRow unit test', () => {
  beforeEach(() => {
    updateGroup.mockResolvedValue(true);
  });

  it('GroupRow Snapshot test', async () => {
    const history = createMemoryHistory();
    let tree;
    await renderer.act(async () => {
      tree = renderer.create(
        <Router history={history}>
          <GroupRow
            number={1}
            groupId={listOfGroupAll[0].groupId}
            groupPic={listOfGroupAll[0].pic}
            groupName={listOfGroupAll[0].groupName}
            member={listOfGroupAll[0].member}
            userStatus={listOfGroupAll[0].userStatus}
          />
        </Router>,
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('GroupRow contains correct element', async () => {
    await act(async () => {
      const history = createMemoryHistory();
      render(
        <Router history={history}>
          <GroupRow
            number={1}
            groupId={listOfGroupAll[0].groupId}
            groupPic={listOfGroupAll[0].pic}
            groupName={listOfGroupAll[0].groupName}
            member={listOfGroupAll[0].member}
            userStatus={listOfGroupAll[0].userStatus}
          />
        </Router>,
      );
    });
    // contains Galaxy Group
    expect(screen.getByText('Galaxy')).toBeInTheDocument();
    // contains leave button
    expect(screen.getByText('LEAVE')).toBeInTheDocument();
  });

  it('Join button is clickable', async () => {
    await act(async () => {
      const history = createMemoryHistory();
      render(
        <Router history={history}>
          <GroupRow
            number={1}
            groupId={listOfGroupAll[0].groupId}
            groupPic={listOfGroupAll[0].pic}
            groupName={listOfGroupAll[0].groupName}
            member={listOfGroupAll[0].member}
            userStatus={listOfGroupAll[0].userStatus}
          />
        </Router>,
      );
    });
    // get button
    const button = screen.getByText('LEAVE');
    // click button
    await act(async () => {
      userEvent.click(button);
    });
    // check button status
    expect(button.innerHTML).toContain('JOIN');
    // click button again
    await act(async () => {
      userEvent.click(button);
    });
    // check button status
    expect(button.innerHTML).toContain('PENDING');
    // click button again
    await act(async () => {
      userEvent.click(button);
    });
    // check button status
    expect(button.innerHTML).toContain('JOIN');
  });

  it('path rendering', async () => {
    let history;
    await act(async () => {
      history = createMemoryHistory();
      render(
        <Router history={history}>
          <GroupRow
            number={1}
            groupId={listOfGroupAll[0].groupId}
            groupPic={listOfGroupAll[0].pic}
            groupName={listOfGroupAll[0].groupName}
            member={listOfGroupAll[0].member}
            userStatus={listOfGroupAll[0].userStatus}
          />
        </Router>,
      );
    });
    // get box
    const box = screen.getByTestId('b1');
    userEvent.click(box);
    expect(history.location.pathname).toBe(`/groupboard/${listOfGroupAll[0].groupId}`);
  });
});
