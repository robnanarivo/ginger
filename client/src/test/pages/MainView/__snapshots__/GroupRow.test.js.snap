// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupRow unit test GroupRow Snapshot test 1`] = `
<div
  className="MuiBox-root css-3cku3k"
>
  <div
    className="MuiBox-root css-11ldntd"
    data-testid="b1"
    onClick={[Function]}
  >
    <div
      className="MuiBox-root css-1q6kmr2"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1pjrok-MuiTypography-root"
      >
        1
      </p>
      <div
        className="MuiBox-root css-i9gxme"
      />
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-9cpfib-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1pjrok-MuiTypography-root"
    >
      Galaxy
    </p>
    <div
      className="MuiBox-root css-i9gxme"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      40 Members
    </p>
  </div>
  <div
    className="MuiBox-root css-i9gxme"
  />
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-n9pybf-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    LEAVE
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
